### PR TITLE
test mne_qt_browser, remove test warning for coverage

### LINKED ISSUE #316: CI failures (linux, multiprocessing, pickle)
https://github.com/autoreject/autoreject/actions/runs/6388600854/job/17338706814#step:11:234

- potentially need to resolve a warning
- potentially need to install mne-qt-browser for not skipping one of the tests
- must resolve the multiprocessing failur which curiously only appears on linux

image

### PR DESCRIPTION
**Pull request by sappelhoff**
- closes #316 

actual CI failure addressed upstream in MNE-Python: 

- https://github.com/mne-tools/mne-python/issues/12062

this one just deals with a warning and with a skipped test.

### CODE REVIEW

**Comment by larsoner** [comment]
> Shouldn't QT bindings be automatically installed with mne_qt_browser?

No because there are 4 choices and we shouldn't choose for you. PyQt6 is best in my opinion though

File: setup.py [hunk 1]
```diff
@@ -61,6 +61,7 @@
           ],
           extras_require={
               'test': [
+                  'mne_qt_browser',
```
Comment (jasmainak) [inline setup.py:64]: looks like we need to add `PyQt5` as a dependency too since the mne_qt_browser could not find Qt bindings

Comment (sappelhoff) [reply setup.py:64]: true, thanks for reminding me :-) pushed a commit!

**Comment by sappelhoff** [comment]
failing due to the current stable version of mne-qt-browser still using a deprecated method from mne. 

Not sure what to do ... wait for a release? Or pin the dev version of mne-qt-browser?

cc @larsoner

**Comment by larsoner** [comment]
Feel free to vote yes in https://github.com/mne-tools/mne-qt-browser/issues/207 and I can quickly cut a release, hopefully today

File: examples/plot_visualize_bad_epochs.py [hunk 2]
```diff
@@ -110,7 +110,7 @@
 import matplotlib.patches as patches  # noqa
 from autoreject import set_matplotlib_defaults  # noqa
 
-set_matplotlib_defaults(plt, style='seaborn-white')
+set_matplotlib_defaults(plt, style='white')
```
Comment (sappelhoff) [inline examples/plot_visualize_bad_epochs.py:113]:
```suggestion
set_matplotlib_defaults(plt, style='default')
```

**Comment by sappelhoff** [comment]
We're getting this error now 

> Cannot connect to a valid display

for Ubuntu only :thinking: 

@larsoner I cannot seem to access the error logs from azure. Is there an org that you can invite me to using my Email: stefan.appelhoff [--at--] mailbox.org ?

**Comment by larsoner** [comment]
> for Ubuntu only 🤔

For this I use (for all OSes):

https://github.com/mne-tools/mne-python/blob/c7c8a2919734963726cd2d04fa9e01002b7f843c/.github/workflows/tests.yml#L86-L89

> @larsoner I cannot seem to access the error logs from azure. Is there an org that you can invite me to using my Email: stefan.appelhoff [--at--] mailbox.org ?

Yes in principle but really I think we should just `rm azure-pipelines.yml` and remove it from the required checks. Looks like it just runs stable and dev and we already do that on GHA. And it's harder to maintain... speaking of which, this is the error:
```
________________________ ERROR at setup of test_viz[qt] ________________________
autoreject/conftest.py:114: in browser_backend
    with use_browser_backend(backend_name) as backend:
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/contextlib.py:135: in __enter__
    return next(self.gen)
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/mne/viz/_figure.py:839: in use_browser_backend
    old_backend = set_browser_backend(backend_name)
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/mne/viz/_figure.py:774: in set_browser_backend
    _load_backend(backend_name)
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/mne/viz/_figure.py:651: in _load_backend
    from mne_qt_browser import _pg_figure as backend
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/mne_qt_browser/_pg_figure.py:46: in <module>
    from qtpy.QtGui import (
/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/qtpy/QtGui.py:57: in <module>
    from PyQt6 import QtGui
E   ImportError: libEGL.so.1: cannot open shared object file: No such file or directory
```
You can get GL support on Windows easily on GHA using the `setup-headless-action` thing I linked to above. It's also possible on Azure but (again) more of a pain:

https://github.com/mne-tools/mne-python/blob/c7c8a2919734963726cd2d04fa9e01002b7f843c/azure-pipelines.yml#L270-L274

**Comment by sappelhoff** [comment]
ok, done -- thanks for the advice and let's see whether this passes now. :-)

**Comment by jasmainak** [comment]
Thanks @sappelhoff !

**Pull request merged**